components/gpio_control/GPIODevices/shutdown_button.py

```diff
@@ -54,8 +54,12 @@ def callbackFunctionHandler(self, *args):
                # leave it on for the moment, it will be off when the system is down
                self.when_pressed(*args)
            else:
-               # switch off LED if pressing was cancelled early (during flashing)
-               self.set_led(GPIO.LOW)
+               # leave LED on if pressing was cancelled early (during flashing) and status led uses the same pin as shutdown led
+               if status_led.GPIO and (status_led.GPIO is self.led_pin):
+                  self.set_led(GPIO.HIGH)
+               else:
+                  self.set_led(GPIO.LOW)
+                  
```

Comment:
This is what I would like to check and I need the status led configuration here, but I don't have it.

Reply:
I cannot simply add the StatusLED config to the[ construction of the ShutdownButton](https://github.com/topas-rec/RPi-Jukebox-RFID/blob/b63b7150bb16c8ddbd39b903cf600d5f09999d7c/components/gpio_control/gpio_control.py#L53), because the StatusLED config might be read from the config file after (is placed below) the shutdown button.

Reply:
Made a dependency check function after all devices have been read in. So it is a kind of late injection of the status that the  shutdown button led should have after a shutdown is cancelled.
https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1997/files#diff-3030f12dc8c3fa2e2c9c64dd6ca0ee1ae9a539a301e58e0211379e89ee17995cR141